src/Ninject.Web.AspNetCore.Test/Unit/DuplicateDescriptorTest.cs

```diff
@@ -3,56 +3,139 @@
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Ninject.Web.AspNetCore.Test.Fakes;
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using Xunit;
 
 namespace Ninject.Web.AspNetCore.Test.Unit
 {
 	public class DuplicateDescriptorTest : TestKernelContext
 	{
-		[Fact]
-		public void ServiceProviderWithDuplicateDescriptorsResolvesLatest()
+		[Theory]
+		[MemberData(nameof(ResolverData))]
+		public void BindingsFromDescriptors_ResolveWarrior_ResolvesLatestWithLatestDependency(Resolver resolver)
 		{
-			var serviceProvider = CreateDuplicateDescriptors().Get<IServiceProvider>();
+			var kernel = CreateDuplicateDescriptors();
 
-			var warrior = serviceProvider.GetService<IWarrior>();
+			var warrior = resolver.Resolve<IWarrior>(kernel);
 			warrior.Should().NotBeNull();
-			warrior.Should().BeOfType<Ninja>();
+			warrior.Should().BeOfType<Knight>();
+			(warrior as Knight).Weapon.Should().BeOfType<Longsword>();
 		}
 
-		[Fact]
-		public void KernelGetWithDuplicateDescriptorsThrows()
+		[Theory]
+		[MemberData(nameof(ResolverData))]
+		public void BindingsFromDescriptors_ResolveAllWarriors_ResolvesAllWithLatestDependency(Resolver resolver)
 		{
 			var kernel = CreateDuplicateDescriptors();
```

Comment:
What about mixed cases? I.e. waepons are added to ServiceDescriptors and warriors to kernel?

Reply:
Added some more test cases for that. They behave as you would "expect".

Reply:
Very good, thanks. Yes, they work as I expect it.